### PR TITLE
ch4/ofi: fix errnames in MPIDI_OFI_CALL

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
OFI call error names are wrapped inside various `MPIDI_OFI_CALL` macros, and cannot be parsed by `maint/extracterrmsgs`, thus they are worked-around with dummy file `src/mpid/ch4/netmod/ofi/catalog.c`. Over the time, we had discrepancies and some of the ofi calls are missing error messages. For example, we are missing `**ofid_tinjectdata`.

Instead of manually maintain them, amend `extracterrmsgs` script to generate errnames based on `MPIDI_OFI_CALL` usages.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/most

File: src/mpid/ch4/netmod/ofi/ofi_impl.h [hunk 1]
```diff
@@ -119,6 +119,27 @@ int MPIDI_OFI_handle_cq_error(int vci, int nic, ssize_t ret);
         } \
     } while (0)
 
+#define MPIDI_OFI_CALL_RETRY_AM(FUNC,vci_,STR)                          \
+    do {                                                                \
+        ssize_t _ret;                                                   \
+        do {                                                            \
+            _ret = FUNC;                                                \
+            if (likely(_ret==0)) break;                                  \
+            MPIDI_OFI_ERR(_ret != -FI_EAGAIN,                  \
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_impl.h:128]: commit message is `remove MPIDI_OFI_ERR` but I think it's still in use?

**Review by raffenet — COMMENTED** [review]
just the one comment/clarification. this is probably worth picking back to 4.2.x, too.

File: src/mpid/ch4/netmod/ofi/ofi_impl.h (hunk 1 above)
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_impl.h:128]: I'll update the commit message

Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_impl.h:128]: Updated to: "ch4/ofi: remove MPIDI_OFI_ERR usages outside ofi_impl.h".

**Pull request merged**